app/services/foreman_ansible/inventory_creator.rb

```diff
@@ -8,11 +8,42 @@ def initialize(hosts)
       @hosts = hosts
     end
 
-    def tempfile
-      tempfile = Tempfile.new("foreman-#{SecureRandom.uuid}-inventory")
-      tempfile.write(hosts.map(&:fqdn).join('\n'))
-      tempfile.close
-      tempfile
+    def json
+      { 'all' => { 'hosts' => hosts.map(&:fqdn) },
+        '_meta' => { 'hostvars' => host_vars } }
```

Comment:
I see, thanks. I've set my variable in the playbook to `foreman['surefail_package']` and it did not work either,  'dict object' has no attribute 'surefail_package'`. 

The `foreman` dict contains this:
```json
    "msg": "From Foreman host {u'comment': u'', u'environment_id': 1, u'managed': False, u'model_id': 2, u'ip': u'192.168.0.106', u'provision_method': u'build', u'last_compile': u'2016-09-07T09:04:05.000Z', u'updated_at': u'2016-09-07T09:04:28.000Z', u'realm_name': None, u'puppet_status': 16781312, u'sp_mac': None, u'operatingsystem_name': u'Fedora release 24 (Twenty Four)', u'disk': None, u'location_id': 1, u'domain_id': 1, u'location_name': u'Spain', u'compute_resource_id': None, u'subnet_id': None, u'sp_name': None, u'environment_name': u'production', u'architecture_name': u'x86_64', u'domain_name': u'lobatolan.home', u'capabilities': [u'build'], u'id': 2, u'organization_name': u'Lobacorp', u'global_status': 2, u'architecture_id': 1, u'build': False, u'hostgroup_title': None, u'subnet_name': None, u'sp_ip': None, u'model_name': u'Satellite Radius P55W-B', u'global_status_label': u'Error', u'owner_id': 4, u'operatingsystem_id': 2, u'sp_subnet_id': None, u'configuration_status_label': u'Error', u'name': u'toshiba.lobatolan.home', u'certname': u'toshiba.lobatolan.home', u'image_name': None, u'organization_id': 2, u'image_id': None, u'mac': u'60:57:18:52:89:97', u'ip6': None, u'puppet_ca_proxy_id': 2, u'last_report': u'2016-09-07T07:03:52.000Z', u'subnet6_id': None, u'puppet_proxy_id': 2, u'compute_profile_name': None, u'uuid': None, u'ptable_name': None, u'image_file': u'', u'compute_resource_name': None, u'created_at': u'2016-08-10T18:30:01.000Z', u'enabled': True, u'hostgroup_id': None, u'configuration_status': 16781312, u'hostgroup_name': None, u'use_image': None, u'compute_profile_id': None, u'medium_id': None, u'subnet6_name': None, u'ptable_id': None, u'owner_type': u'User', u'realm_id': None, u'installed_at': None, u'medium_name': None}"
```

and `foreman_params` contains this:

```json
    "msg": "From Foreman host [{u'id': 10, u'value': u'tig', u'name': u'surefail_package'}]"
```

The latter is fine, it's *almost*how the inventory does it. It should be:

```json
 { 'host_param_name' : 'host_param_value', 'host_param2_name' : 'host_param2_value' }
```

It's a small change but with that the inventory and foreman_params have the same format and playbooks can be reused.

See https://github.com/theforeman/foreman_ansible_inventory#variables-and-parameters